router/pkg/config/config_test.go

```diff
@@ -752,3 +754,307 @@ headers:
 	})
 
 }
+
+func TestConfigMerging(t *testing.T) {
```

Comment:
Please improve the `router/docs/Configuration.md` that we should prefer `map` usage where possible to make it easier for users to merge configuration. You can highlight on the merge behaviour. Is not desired to use a list when a map is sufficient.